### ISSUE TITLE
[Bug]:  row index not being updated on sorting

### ISSUE THREAD

**Issue by shrilakshmishastry** [ceolter/ag-grid#10798]
### Link to reproducible scenario

https://www.ag-grid.com/examples/getting-started/quick-start-example/reactFunctional/

### Describe the bug

Steps to reproduce 
1. Sort the table 
2. Try to access it using VoiceOver and ctrl+option+ arrows(ROTAR)

Actual behaviour
The rows are not in  the same order as they appear on screen 

Expected behaviour
The rows should be in the same order as they appear on screen 

More information
https://drive.google.com/file/d/1F0NNVN6QHVqtaLm26EaAJrv9OeDRWrY3/view?usp=sharing

I want to fix the issue. I tried to set up the project locally, but was unable to do so. Any doc to setup project  would be helpful as I am testing ag-grid for accessibility 

### Version

32.3.8

### Does the issue occur for a specific framework only?

All languages

### Is the issue only observable on a specific browser?

All browsers

**Comment by ivan-nejezchleb** [comment]
We have the same issue but not caused by sorting but scrolling:
Open eg. **https://www.ag-grid.com/angular-data-grid/tool-panel-columns/**

Scroll table few times by PageDown and then return back to top by PageUp. 
Check the DOM. Rows are in dom in different order than how they are visible in table. This prevents screen reader to navigate predictably

<img width="1777" height="964" alt="Image" src="https://github.com/user-attachments/assets/afff640e-05ef-485c-9a0e-705251360308" />

**Comment by StephenCooper** [comment]
> We have the same issue but not caused by sorting but scrolling: Open eg. **https://www.ag-grid.com/angular-data-grid/tool-panel-columns/**
> 
> Scroll table few times by PageDown and then return back to top by PageUp. Check the DOM. Rows are in dom in different order than how they are visible in table. This prevents screen reader to navigate predictably
> 

Please see https://ag-grid.com/react-data-grid/accessibility/#customising-the-grid-for-accessibility
